src/txn/txn.c

```diff
@@ -1894,9 +1894,9 @@ __wt_txn_commit(WT_SESSION_IMPL *session, const char *cfg[])
             break;
         }
 
-        /* If we used the cursor to resolve prepared updates, the key now has been freed. */
+        /* If we used the cursor to resolve prepared updates, free and clear the key. */
         if (cursor != NULL)
-            WT_CLEAR(cursor->key);
+            __wt_buf_free(session, &cursor->key);
```

Comment:
For re-reviewer(s): I changed these lines in txn.c.  Doing a wholesale clear of the key would lose any storage in `cursor->key.mem` (a memory leak).  `__wt_buf_free` frees the storage, and also clears the `WT_ITEM`.